internal/services/dns/dns_zone_resource.go

```diff
@@ -4,377 +4,364 @@
 package dns
 
 import (
+	"context"
 	"fmt"
 	"strings"
 	"time"
 
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-helpers/lang/response"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
-	"github.com/hashicorp/go-azure-helpers/resourcemanager/tags"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/dns/2018-05-01/recordsets"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/dns/2018-05-01/zones"
-	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/dns/migration"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/dns/validate"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
-	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
-	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
-func resourceDnsZone() *pluginsdk.Resource {
-	return &pluginsdk.Resource{
-		Create: resourceDnsZoneCreateUpdate,
-		Read:   resourceDnsZoneRead,
-		Update: resourceDnsZoneCreateUpdate,
-		Delete: resourceDnsZoneDelete,
+var (
+	_ sdk.Resource                   = DnsZoneResource{}
```

Comment:
It looks like this line is unneeded as `ResourceWithUpdate` and `ResourceWithStateMigration` encapsulate `Resource` so I believe this can be removed. Do you mind confirming?

Reply:
You're right the `ResourceWithStateMigration` and `ResourceWithUpdate` types both extend `Resource` so that line can be removed. 
I left it in as a few other resources I used as reference also have both so I assumed it was a style choice to be explicit.
Happy to take it out